doc/whats_new/v1.1.rst

```diff
@@ -181,6 +181,10 @@ Changelog
 :mod:`sklearn.ensemble`
 .......................
 
+- |Enhancement| :class:`BaseGradientBoosting` now calls :meth:`base.BaseEstimator._validate_data` with
+  parameter `force_all_finite` equals to False when it is already initialized.
+  :pr:`22089` by :user:`Geoffrey Paris <Geoffrey-Paris>`.
```

Comment:
Since the changelog is public facing, it is best to list out all the estimators that is affected by the change:

```rst
- |Efficiency| :class:`GradientBoostingClassifier` and :class:`GradientBoostingRegressor`
  does not check if `X` is finite during the second call to `fit` when the estimator was
  warm started. :pr:`22089` by :user:`Geoffrey Paris <Geoffrey-Paris>`.
```

(Also we try not to mention private methods such as `_validate_data`)

Reply:
Adressed by c9be126